worksheet.rb

```diff
@@ -23,12 +38,209 @@
+            date: '4th Feb 2016',
+            cost: 5,
+            rider_id: 'RD0066',
+            rating: 5
+        },
+        {
+            driver_id: 'DR0004',
+            date: '4th Feb 2016',
+            cost: 10,
+            rider_id: 'RD0022',
+            rating: 4
+        },
+        {
+            driver_id: 'DR0002',
+            date: '4th Feb 2016',
+            cost: 15,
+            rider_id: 'RD0013',
+            rating: 1
+        },
+        {
+            driver_id: 'DR0003',
+            date: '5th Feb 2016',
+            cost: 50,
+            rider_id: 'RD0003',
+            rating: 2
+        },
+        {
+            driver_id: 'DR0002',
+            date: '5th Feb 2016',
+            cost: 35,
+            rider_id: 'RD0066',
+            rating: 3
+        },
+        {
+            driver_id: 'DR0004',
+            date: '5th Feb 2016',
+            cost: 20,
+            rider_id: 'RD0073',
+            rating: 5
+        },
+        {
+            driver_id: 'DR0001',
+            date: '5th Feb 2016',
+            cost: 45,
+            rider_id: 'RD0003',
+            rating: 2
+        }
+    ]
+}
+
 ########################################################
 # Step 4: Total Driver's Earnings and Number of Rides
 
 # Use an iteration blocks to print the following answers:
 # - the number of rides each driver has given
+def num_rides(structured_data)
+  driver_ride_count = {}
+  structured_data[:rides].each do |ride|
+    if driver_ride_count.include?(ride[:driver_id])
+      driver_ride_count[ride[:driver_id]] += 1
+    else
+      driver_ride_count[ride[:driver_id]] = 1
+    end
+  end
+  return driver_ride_count
+end
+
+p "Number of rides completed per driver:"
+p num_rides(ride_share_data)
+
+
 # - the total amount of money each driver has made
+def driver_earnings(structured_data)
+  earnings = {}
+  structured_data[:rides].each do |ride|
+    if earnings.include?(ride[:driver_id])
+      earnings[ride[:driver_id]] += ride[:cost]
+    else
+      earnings[ride[:driver_id]] = ride[:cost]
+    end
+  end
+  return earnings
+end
+
+p "Total cost per driver:"
+p driver_earnings(ride_share_data)
+
+
 # - the average rating for each driver
+def average_rating(structured_data)
+  avg_driver_ratings = {}
+
+  structured_data[:rides].each do |ride|
+    if avg_driver_ratings.include?(ride[:driver_id])
+      avg_driver_ratings[ride[:driver_id]] << ride[:rating].to_f
+    else
+      avg_driver_ratings[ride[:driver_id]] = [ride[:rating].to_f]
+    end
+  end
+
+  avg_driver_ratings.each do |driver_id, ratings|
+    avg_driver_ratings[driver_id] = (ratings.sum / ratings.size).round(1)
+  end
+
+  return avg_driver_ratings
+end
+
+p "Average rating per driver:"
+p average_rating(ride_share_data)
+
+
 # - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+def max_earner(structured_data)
+  earnings = driver_earnings(structured_data)
```

Comment:
Great work using the data generated from `driver_earnings` to help with this calculation.